tests/integration/src/test/scala/mesosphere/marathon/integration/ResidentTaskIntegrationTest.scala

```diff
@@ -97,6 +97,41 @@ class ResidentTaskIntegrationTest extends AkkaIntegrationTest with EmbeddedMarat
       waitForStatusUpdates(StatusUpdate.TASK_FINISHED)
     }
 
+    "reattach resident task after task failure" in new Fixture {
+      Given("An app that writes into a persistent volume")
+      val containerPath = "persistent-volume"
+      val id = appId("resident-task-with-persistent-volume-will-reattach-after-failure")
+      val cmd = s"""echo hello >> $containerPath/data && ${appMockCmd(id, "v1")}"""
+      val healthCheck = appProxyHealthCheck().copy(path = Some("/ping"))
+      val app = residentApp(id = id, containerPath = containerPath, cmd = cmd, portDefinitions = Seq(PortDefinition(name = Some("http"))))
+        .copy(networks = Seq(Network(mode = NetworkMode.Host)), backoffSeconds = 1, healthChecks = Set(healthCheck))
+
+      When("a task is launched")
+      createSuccessfully(app)
+
+      And("the app dies")
+      val tasks = marathon.tasks(id).value
+      tasks should have size (1)
+      val failedTask = tasks.head
+      val failedAppMock = AppMockFacade(failedTask)
+      failedAppMock.get(s"/$containerPath/data").futureValue should be("hello\n")
+      failedAppMock.suicide().futureValue
+
+      Then("the failed task is restarted")
+      val newTask = eventually {
+        val newTasks = marathon.tasks(id).value
+        newTasks should have size (1)
+        val newTask = newTasks.head
+        newTask.state should be("TASK_RUNNING")
+        newTask.healthCheckResults.head.alive should be(true)
```

Comment:
This is fun. The health check is not reset after a failover. At least it seems like that since the last build failed on `AppMockFacade(newTask).get(s"/$containerPath/data").futureValue should be("hello\nhello\n")` below with

```
akka.stream.StreamTcpException, with message: Tcp command [Connect(172.16.10.234:36342,None,List(),Some(10 seconds),true)] failed because of Connection refused.
``` 

which means the new task was not healthy yet. The sandbox logs suggest that the app was not even pinged.